presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetCompressionUtils.java

```diff
@@ -45,23 +47,22 @@ public static Slice decompress(CompressionCodecName codec, Slice input, int unco
             case GZIP:
                 return decompressGzip(input, uncompressedSize);
             case SNAPPY:
-                return decompressSnappy(input, uncompressedSize);
+                return decompress(new SnappyDecompressor(), input, uncompressedSize);
+            case LZO:
+                return decompress(new LzoDecompressor(), input, uncompressedSize);
```

Comment:
So I think I understand the problem.  The code you provided is using the Hadoop wrapper around the real LZO algorithm.  Then the code attempts to the Airlift version of the LZO algorithm.  This doesn't work because you are missing the code that removes the Hadoop framing around the LZO data.  To do this you should be able to call `new LzoCodec().createDecompressor()` to get a Hadoop Deecompressor, which understands the Hadoop framing.  Unfortunately, I did not implement the Decompressor (or Compressor) interfaces for our codecs because I could not find any code in Hadoop that actually used them.  Ideally, the authors of Parquet would have used the raw algorithm directly, but instead used the Hadoop wrappers (which are poorly designed).  

If you need these interfaces implemented for this code, I can add them.